plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs

```diff
@@ -335,10 +322,6 @@ data Ctrex
   | CtrexTermEvaluationFail
     ClosedTypeG
     ClosedTermG
-  | CtrexTermEvaluationMismatch
-    ClosedTypeG
-    ClosedTermG
-    [Term TyName Name DefaultUni DefaultFun ()]
```

Comment:
I am surprised this has gone. Don't we need it to test production-type-CK against agda-typed-blah?

Reply:
For some reason I thought we only case about NEAT stuff inside `plutus-core`. Now I see (after looking at errors in CI) that this is not true. I'll fix that.

Reply:
There's some confusing dependency management going on here. Here's my understanding:
1. plutus-metatheory depends on plutus-core for parsing and pretty printing etc.
2. Both plutus-core and plutus-metatheory use NEAT. So, the generators need to be part of plutus-core as we can't depend on tests from another package - annoyingly putting testing code into the library itself. 
3. plutut-core-test uses NEAT to test plutus-core stuff only.
4. plutus-metatheory's tests actually test plutus-metatheory's PLC implementation against plutus-core's.

Reply:
I see, thanks. I've returned the constructor, tests pass.